components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/util/SAMLFedLogoutUtil.java

```diff
@@ -0,0 +1,317 @@
+            logoutResp.setVersion(SAMLVersion.VERSION_20);
+            logoutResp.setStatus(buildStatus(statusCode, statusMsg));
+            logoutResp.setIssueInstant(new DateTime());
+            logoutResp.setDestination(acsUrl);
+
+            if (TRUE.equals(isResponseSigned) && SUCCESS_CODE.equals(statusCode)) {
+                SSOUtils.setSignature(logoutResp, null, null, isIncludeCert,
+                        new X509CredentialImpl(samlMessageContext.getTenantDomain(), null));
+            }
+            return logoutResp;
+
+        } catch (SAMLSSOException e) {
+            throw new SAMLIdentityException("Error occurred while setting the signature of logout response", e);
+        } catch (FrameworkException e) {
+            throw new SAMLIdentityException("Error in bootstrapping the OpenSAML2 library", e);
+        } catch (IdentityException e) {
+            throw new SAMLIdentityException("Error while building Secure Random ID for the logout response", e);
+        }
+    }
+
+    /**
+     * Build the issuer for the logout response.
+     *
+     * @param issuerID Index of the issuer in the SAML Logout Request.
+     * @return Built issuer object for Logout Response.
+     */
+    private static Issuer getIssuer(String issuerID) {
+
+        IssuerBuilder issuerBuilder = new IssuerBuilder();
+        Issuer issuer = issuerBuilder.buildObject();
+        issuer.setValue(issuerID);
+        return issuer;
+    }
+
+    /**
+     * Validate the signature of the LogoutRequest against the given certificate.
+     *
+     * @param logoutRequest {@link LogoutRequest}object to be validated.
+     * @return true           If signature of the Logout Request is valid.
+     * @throws SAMLIdentityException Error when validating the signature.
+     */
+    public static boolean isValidSignature(LogoutRequest logoutRequest, SAMLMessageContext
+            samlMessageContext) throws SAMLIdentityException {
+
+        String issuer = logoutRequest.getIssuer().getValue();
+        X509Certificate x509Certificate = generateX509Certificate(samlMessageContext.
+                getFederatedIdP().getCertificate());
+
+        LogoutReqSignatureValidator signatureValidator = new LogoutReqSignatureValidator();
+        try {
+            if (samlMessageContext.getSAMLLogoutRequest().isPost()) {
+                return signatureValidator.validateXMLSignature(logoutRequest, (X509Credential) x509Certificate, null);
+
+            } else {
+                return signatureValidator.validateSignature(samlMessageContext.getSAMLLogoutRequest().getQueryString(),
+                        issuer, x509Certificate);
+            }
+        } catch (SecurityException | IdentityException e) {
+            String message = "Process of validating the signature failed for the logout request with issuer: "
+                    + logoutRequest.getIssuer().getValue();
+            if (log.isDebugEnabled()) {
+                log.debug(message, e);
+            }
```

Comment:
Why do we need this debug log if we are throwing this?

Reply:
will remove it.